move-text-tests.el

```diff
@@ -0,0 +1,111 @@
+;;; move-text-tests.el --- Tests for Move-Text -*- lexical-binding: t; -*-
+;;
+;; Created: September 07, 2022
+
+;;; Code:
+
+(require 'move-text)
+(require 'ert)
+
+(defmacro should-on-temp-buffer (input-string expect-string &rest body)
+  "INPUT-STRING processed by BODY forms in a temp buffer should equal EXPECT-STRING."
+  (declare (indent 1) (debug t))
+  `(should (string= ,expect-string
+                    (with-temp-buffer
+                    ;; (with-current-buffer-window "*ERT-should-temp-buffer*"
+                    ;;   (erase-buffer)
+                      (insert ,input-string)
+                      (goto-char (point-min))
+                      ,@body
+                      (buffer-string)))))
+
+(let ((transient-mark-mode t))
+
+  (ert-deftest move-text-down-test ()
+    "Move text down by (1) one line, (2) by region."
+    (should-on-temp-buffer
+        "This is a test
+Line 2
+Line 3
+"
+      "Line 2
+This is a test
+Line 3
+"
+      (call-interactively #'move-text-down))
+
+;;     (should-on-temp-buffer
+;;         "This is a test
+;; Line 2
+;; Line 3
+;; Line 4
+;; Line 5
+;; Line 6
+;; "
+;;         "This is a test
+;; Line 2
+;; Line 4
```

Comment:
and the region test?

Reply:
(these line move tests were already working...)

Reply:
I've split up between line and region tests. So currently it is four tests